pennylane/devices/default_qubit_jax.py

```diff
@@ -0,0 +1,193 @@
+    supports classical backpropagation as a means to compute the gradient. This can be faster than
+    the parameter-shift rule for analytic quantum gradients when the number of parameters to be
+    optimized is large.
+
+    To use this device, you will need to install jax:
+
+    .. code-block:: console
+
+        pip install jax jaxlib
+
+    **Example**
+
+    The ``default.qubit.jax`` is designed to be used with end-to-end classical backpropagation
+    (``diff_method="backprop"``) with the jax interface. This is the default method of
+    differentiation when creating a QNode with this device.
+
+    Using this method, the created QNode is a 'white-box', and is
+    tightly integrated with your jax computation:
+
+    >>> dev = qml.device("default.qubit.jax", wires=1)
+    >>> @qml.qnode(dev, interface="jax", diff_method="backprop")
+    ... def circuit(x):
+    ...     qml.RX(x[1], wires=0)
+    ...     qml.Rot(x[0], x[1], x[2], wires=0)
+    ...     return qml.expval(qml.PauliZ(0))
+    >>> weights = jnp.array([0.2, 0.5, 0.1])
+    >>> grad_fn = jax.grad(circuit)
+    >>> print(grad_fn(weights))
+    array([-2.2526717e-01 -1.0086454e+00  1.3877788e-17])
+
+    There are a couple of things to keep in mind when using the ``"backprop"``
+    differentiation method for QNodes:
+
+    * You must use the ``"jax"`` interface for classical backpropagation, as jax is
+      used as the device backend.
+
+    * Only exact expectation values, variances, and probabilities are differentiable.
+      When instantiating the device with ``analytic=False``, differentiating QNode
+      outputs will result in an error.
+
+    Args:
+        wires (int): the number of wires to initialize the device with
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to 1000 if not specified.
+            If ``analytic == True``, then the number of shots is ignored
+            in the calculation of expectation values and variances, and only controls the number
+            of samples returned by ``sample``.
+        analytic (bool): Indicates if the device should calculate expectations
+            and variances analytically. In non-analytic mode, the ``diff_method="backprop"``
+            QNode differentiation method is not supported and it is recommended to consider
+            switching device to ``default.qubit`` and using ``diff_method="parameter-shift"``.
+    """
+
+    name = "Default qubit (jax) PennyLane plugin"
+    short_name = "default.qubit.jax"
+
+    parametric_ops = {
+        "PhaseShift": jax_ops.PhaseShift,
+        "RX": jax_ops.RX,
+        "RY": jax_ops.RY,
+        "RZ": jax_ops.RZ,
+        "Rot": jax_ops.Rot,
+        "CRX": jax_ops.CRX,
+        "CRY": jax_ops.CRY,
+        "CRZ": jax_ops.CRZ,
+        "MultiRZ": jax_ops.MultiRZ,
+    }
+
+    C_DTYPE = jnp.complex64
+    R_DTYPE = jnp.float32

```

Comment:
Lower precision runs faster on JAX

Reply:
We did the same thing in the Strawberry Fields TF backend (`tf.complex64`), and later realized due to user bug reports how many edge cases would propagate the numerical error until it was no longer negligible. It might be worth adding a way to instantiate this device with higher precision (if the user wants).

Reply:
Agreed, but we should do this across the codebase

Reply:
I'll bump it back to higher precision and leave the tunable precision to a later PR.